### ISSUE TITLE
Handling of common measurement units

### ISSUE THREAD

**Issue by ryanlaclair** [BeamIO-Inc/resippy#16]
It would be great to expand the handling of units conversion for concrete implementations of our abstract base classes.

An example of an an existing `init` method:
```python
def init_from_length(self, length_meters):
    self.length_meters = length_meters
```

Instead this could be changed to:
```python
def init_from_length(self, length, length_unit):
    # immediately convert to common meters unit?
    # store length and length_unit and handle conversion later?
    pass
```

**Comment by jcasey-beamio** [comment]
I like the idea of doing something like this.  Wondering if maybe we do something similiar to our point calculators, where a pointCalc object has "native" units?  Not sure if we should automatically do a conversion during an init step, I like the idea of carrying units around with an object we're initializing.

**Comment by jcasey-beamio** [comment]
This was implemented in for some init methods in 0.2.1, should add more of this functionality in more init methods moving forward.
